File: AndroidLearnProject/app/src/main/java/com/android/learn/mcontract/ProjectContract.java
package com.android.learn.mcontract;

import com.android.learn.base.mmodel.ProjectListData;
import com.android.learn.base.mview.BaseView;

/**
 * Created by gaolei on 2018/6/18.
 */

public class ProjectContract {

    public interface Presenter {

        void getProjectInfo(int page, int cid);

        void onRefreshMore(int cid);

        void onLoadMore(int cid);
    }

    public interface View extends BaseView {

        void showProjectList(ProjectListData itemBeans, boolean isRefresh);
    }
}

File: AndroidLearnProject/app/src/main/java/com/android/learn/mcontract/ProjectContract.kt
package com.android.learn.mcontract

import com.android.learn.base.mmodel.ProjectListData

/**
 * Created by gaolei on 2018/6/18.
 */

class ProjectContract {

    interface Presenter {

        fun getProjectInfo(page: Int, cid: Int)

        fun onRefreshMore(cid: Int)

        fun onLoadMore(cid: Int)
    }

    interface View {

        fun showProjectList(itemBeans: ProjectListData, isRefresh: Boolean)
    }
}
